src/HotChocolate/Data/src/Data/Filters/Types/RelayIdFilterFieldExtensions.cs

```diff
@@ -0,0 +1,64 @@
+using System;
+using HotChocolate.Configuration;
+using HotChocolate.Internal;
+using HotChocolate.Types;
+using HotChocolate.Types.Descriptors;
+using HotChocolate.Types.Relay;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace HotChocolate.Data.Filters
+{
+    internal static class RelayIdFilterFieldExtensions
+    {
+        public static IFilterOperationFieldDescriptor ID<TEntityType, TIdType>(
+            this IFilterOperationFieldDescriptor descriptor)
+        {
+            if (descriptor is null)
+            {
+                throw new ArgumentNullException(nameof(descriptor));
+            }
+
+            descriptor.Extend()
+                .OnBeforeCompletion(
+                    (c, d) => AddSerializerToInputField<TEntityType, TIdType>(c, d));
+
+            return descriptor;
+        }
+
+        private static void AddSerializerToInputField<TEntityType, TIdType>(
+            ITypeCompletionContext completionContext,
+            FilterOperationFieldDefinition definition)
+        {
+            ITypeInspector typeInspector = completionContext.TypeInspector;
+            IExtendedType? resultType;
+
+            resultType = typeInspector.GetReturnType(typeof(TIdType[]), true);
+            ExtendedTypeReference resultTypeReference =
+                typeInspector.GetReturnTypeRef(typeof(TEntityType));
+
+            definition.Formatters.Add(
+                CreateSerializer(
+                    completionContext,
+                    resultType,
+                    resultTypeReference));
+        }
+
+        private static IInputValueFormatter CreateSerializer(
+            ITypeCompletionContext completionContext,
+            IExtendedType resultType,
+            ExtendedTypeReference extendedTypeReference)
+        {
+            NameString typeName =
+                completionContext.GetType<IType>(extendedTypeReference).TypeName();
+            IIdSerializer serializer =
+                completionContext.Services.GetService<IIdSerializer>() ??
+                new IdSerializer();
+
+            return new GlobalIdInputValueFormatter(
```

Comment:
@michaelstaib  i cannot make the constructor internal as i use it here. 
How far are you and @benmccallum  on the input value formatter? can we then make it public?

Reply:
I've got no objection to it being public, I could unit test it then for the stuff I added.

My PR is done, just waiting on a review and merge; though my tests add a type interceptor and a lot of stuff that's irrelevant to what I'm testing. If this was public I'd simply all that.